### PR TITLE
feat: add secret push format to AWS secrets manager

### PR DESCRIPTION
**Pull request by Skarlso**
## Problem Statement

What is the problem you're trying to solve?

## Related Issue

Fixes https://github.com/external-secrets/external-secrets/issues/1816

## Proposed Changes

How do you like to solve the issue and why?

## Checklist

- [ ] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [ ] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [ ] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -413,3 +419,45 @@ func ConvertToType[T any](obj interface{}) (T, error) {
...
+	m := map[string]any{}
+	if err := json.Unmarshal(data.Raw, &m); err != nil {
+		return t, fmt.Errorf("failed to parse JSON raw data: %w", err)
+	}
+
+	v, err := dig[T](key, m)
+	if err != nil {
+		if errors.Is(err, errKeyNotFound) {
+			return def, nil
+		}
```
Comment (shuheiktgw) [inline pkg/utils/utils.go:440]: I believe you forgot to return an error here.

File: pkg/provider/aws/secretsmanager/secretsmanager.go [hunk 2]
```diff
@@ -40,6 +40,14 @@ import (
...
 	"github.com/external-secrets/external-secrets/pkg/metrics"
 	"github.com/external-secrets/external-secrets/pkg/provider/aws/util"
+	"github.com/external-secrets/external-secrets/pkg/utils"
+)
+
+// Declares metadata information for pushing secrets to AWS Secret Store.
+const (
+	SecretPushFormatKey    = "secretPushFormat"
+	SecretPushFormatString = "string"
+	SecretPushFormatBinary = "binary"
```
Comment (shuheiktgw) [inline pkg/provider/aws/secretsmanager/secretsmanager.go:50]: Could you add documentation on these metadata?

**Review by shuheiktgw — CHANGES_REQUESTED** [review]
I left a few comments but overall LGTM 🙂

File: pkg/utils/utils.go (hunk 1 above)
Comment (Skarlso) [reply pkg/utils/utils.go:440]: The intention here is that if the key is not found we return the set default value. Arguably I could have named the variable better. 🤣 or at least add a comment.

File: pkg/provider/aws/secretsmanager/secretsmanager.go (hunk 2 above)
Comment (Skarlso) [reply pkg/provider/aws/secretsmanager/secretsmanager.go:50]: Ah thanks. I forgot! 🤦

File: pkg/utils/utils.go (hunk 1 above)
Comment (shuheiktgw) [reply pkg/utils/utils.go:440]: Ah, I see what you are trying to do here 👍

**Pull request merged**